packages/ui/src/hooks/use-apple-auth.ts

```diff
@@ -0,0 +1,92 @@
+import camelcaseKeys from 'camelcase-keys';
+import { useNavigate } from 'react-router-dom';
+
+import { inOperator } from '@/utils';
+
+export const loadAppleSdk = async () =>
+  new Promise((resolve, reject) => {
+    const script = document.createElement('script');
+
+    script.addEventListener('load', resolve);
+    script.addEventListener('error', reject);
+    // eslint-disable-next-line @silverhand/fp/no-mutation
+    script.src =
+      'https://appleid.cdn-apple.com/appleauth/static/jsapi/appleid/1/en_US/appleid.auth.js';
+
+    document.head.append(script);
+  });
+
+export const isAppleConnector = ({ target }: { target: string }) => target === 'apple';
+
+// Derived from https://developer.apple.com/documentation/sign_in_with_apple/sign_in_with_apple_js/configuring_your_webpage_for_sign_in_with_apple
+
+type AppleIdAuth = {
+  auth: {
+    init(
+      config: {
+        clientId?: string;
+        scope?: string;
+        redirectURI?: string;
+        state?: string;
+        nonce?: string;
+        usePopup: boolean;
+      } & Record<string, unknown>
+    ): void;
+    signIn(): Promise<Record<string, unknown>>;
+  };
+};
+
+declare const AppleID: AppleIdAuth | undefined;
+
+export const getAppleSdk = () => {
+  if (AppleID === undefined) {
+    throw new Error('AppleID auth SDK not found.');
+  }
+
+  return AppleID;
+};
+
+const useAppleAuth = () => {
+  const navigate = useNavigate();
+
+  const auth = async (connectorId: string, redirectUri: string) => {
+    const url = new URL(redirectUri);
+    const { redirect_uri: redirectURI, ...rest } = Object.fromEntries(url.searchParams);
```

Comment:
nit: have a util method `parseQueryParameters`

Reply:
didn't notice this. will update soon